docs/service-meshes.asciidoc

```diff
@@ -0,0 +1,295 @@
+
+Create the `elastic-system` namespace and enable sidecar injection:
+
+[source,sh]
+----
+kubectl create namespace elastic-system
+kubectl label namespace elastic-system istio-injection=enabled
+----
+
+Install ECK:
+
+[source,sh,subs="attributes"]
+----
+kubectl apply -f https://download.elastic.co/downloads/eck/{eck_version}/all-in-one.yaml
+----
+
+Check the configuration to ensure that the installation has been successful:
+
+[source,sh]
+----
+kubectl get pod elastic-operator-0 -n elastic-system -o=jsonpath='{range .spec.containers[*]}{.name}{"\n"}'
+----
+
+If the output of the above command contains both `manager` and `istio-proxy`, ECK has been successfully installed with the Istio sidecar injected.
+
+
+[float]
+[id="{p}-service-mesh-istio-stack-connection"]
+==== Connecting Elastic Stack applications to the Istio service mesh
+
+NOTE: This section assumes that you are deploying Elasticsearch, Kibana, or APM Server resources to a namespace that has link:https://istio.io/docs/setup/additional-setup/sidecar-injection/#automatic-sidecar-injection[automatic sidecar injection] enabled.
+
+
+If you have configured Istio in link:https://istio.io/docs/concepts/security/#permissive-mode[permissive mode], examples defined elsewhere in the ECK documentation will continue to work without requiring any modifications. However, if you have enabled strict mutual TLS authentication between services either via global (`MeshPolicy`) or namespace-level (`Policy`) configuration, the following modifications to the resource manifests are necessary for correct operation.
+
+[float]
+[id="{p}-service-mesh-istio-elasticsearch"]
+===== Elasticsearch
+
+[source,yaml,subs="attributes,callouts"]
+----
+apiVersion: elasticsearch.k8s.elastic.co/{eck_crd_version}
+kind: Elasticsearch
+metadata:
+  name: elastic-istio
+spec:
+  version: {version}
+  http:
+    tls: <1>
+      selfSignedCertificate:
+        disabled: true
+  nodeSets:
+  - name: default
+    count: 3
+    config:
+      node.store.allow_mmap: false
+    podTemplate:
+      metadata:
+        annotations:
+          traffic.sidecar.istio.io/excludeOutboundPorts: "9300" <2>
+          traffic.sidecar.istio.io/excludeInboundPorts: "9300"
+----
+
+<1> Disable the default self-signed certificate generated by the operator and allow TLS to be managed by Istio.
+
+<2> Exclude the transport port (port 9300) from being proxied. Currently ECK does not support switching off X-Pack security and TLS for the Elasticserarch transport port. If Istio is allowed to proxy the transport port, the traffic will be encrypted twice and communication between Elasticsearch nodes will be disrupted.
```

Comment:
Maybe remove `Currently` which seems to indicate we have plans to change things in the future.

Reply:
It's something I am investigating at the moment. I don't see a reason why we can't allow TLS to be switched off for the transport port as well. I'll bring it up for discussion once I have more information.